### ISSUE TITLE
False positives

### ISSUE THREAD

**Issue by kvz** [wooorm/alex#33]
Thanks for this tool! I think it will be wonderful to run our blog posts & documentation through this in our CI before deployment.

However, I have the case where

```
Authentication is disabled by default
```

red flags on `disabled`

```
1. [Osama Mahmood](https://twitter.com/OsamaMahmood007) (1)
```

red flags `Osama`

and

```
Felix, (alumnus of both Transloadit and Node.js nowadays
```

red flags `alumnus`

The documents with the false positives are fairly big and subject to change, so Ideally I'd have a whitelist of sorts to mark these exceptions - and still be able to fail on new violations.

What's the best way to achieve this?

**Comment by wooorm** [comment]
Thanks!

They could be detected by #16 and #18, probably.
Also: it’s a really personal question, some people would still prefer `graduate` over `alumnus` in the last example, and the same goes for `disabled` and `turned off` in the first.
Your second example could probably be ignored by alex itself because it’s (clearly) a name.

I always meant for alex to be a help when writing, not in a CI to block publishing. Humans are smarter than alex. But, I do see that people (want to) use it that way though.

Short version: #16 and #18 need to be fixed!

**Comment by joshhunt** [comment]
> Alex: Catch insensitive, inconsiderate writing

`Authentication is disabled by default` isn't insensitive, inconsiderate writing.

**Comment by jdalton** [comment]
>> Alex: Catch insensitive, inconsiderate writing

> Authentication is disabled by default isn't insensitive or inconsiderate writing.

Alex reports it --> may <-- be insensitive.

```bash
warning  `disabled` may be insensitive, use `person with disabilities` instead
```

I think the tool qualifies the statement enough here. Besides the occasional false positive it does happen to catch insensitive, inconsiderate writing.

**Comment by wooorm** [comment]
Thanks @jdalton, I agree with you! Alex isn’t as smart as a human, but it tries its best and is sometimes overly happy to let you know something **may** be insensitive.

**Comment by ixti** [comment]
I guess it was overexcited today for the company I work for, and actually made REALLY terribly offensive proposal:

![20151007_05 24 53_selection 001_39c2](https://cloud.githubusercontent.com/assets/94782/10328366/f7ead658-6cb3-11e5-95e9-d213e6a3c396.png)

PLEASE REMOVE IT!!! THAT'S NOT F**KING APPROPRIATE AT ALL!

**Comment by wooorm** [comment]
@ixti I can see how the bot seems offensive here. However, “pancake face” is an actual ethnic slur (according to [WikiPedia](https://en.wikipedia.org/wiki/List_of_ethnic_slurs#P)). Therefore, I think alex is operating according to its byline, “Catch insensitive, inconsiderate writing”, by warning about it.

Please open a new issue with more information if you’d like to discuss this further, or discus the slackbot on the keoghpe/alex-slack#1.

**Comment by yoshuawuyts** [comment]
@ixti Welp, covering all possible contexts of a slur is hard. It correctly caught "pancake face" as being used as a derogatory term; the suggestion could probably use some work though. Could you clarify in what context it was used so we can add an appropriate suggestion?

Also: I understand that you're upset, but please refrain from dropping all-caps f-bombs in issues. Everyone involved with Alex does so in their spare time and has the best intentions. It'd be appreciated if you'd be considerate of that in future interactions.

**Comment by ixti** [comment]
I was probably a bit over-raged. Sorry guys. It's not issue with alex - it's issue with integration (chat bot). Feel free to delete my comments here, as they are really not about alex but about chat bot that uses alex.

**Comment by ixti** [comment]
I apologize for my pretty rude and aggressive comment before (deleted it myself).
In that particular case issue was definitely not with alex but with integration itself.

**Comment by wooorm** [comment]
Thanks for being so considerate @ixti :smile:

**Comment by wooorm** [comment]
Been a while. Thanks for hanging on. The just releases [`2.0.0`](https://github.com/wooorm/alex/releases/tag/2.0.0) version fixes this by introducing several mechanisms to control the messages suggested by **alex**.

**Comment by kvz** [comment]
Awesome!
